src/vip_ipykernel/vip_ipykernel_launcher.py

```diff
@@ -0,0 +1,46 @@
+"""Taken and modified from
+https://github.com/ipython/ipykernel/blob/master/ipykernel_launcher.py
+
+Entry point for launching an ViP-IPython kernel.
+"""
+
+import subprocess
+import sys
+from pathlib import Path
+
+VENV_NAMES = ['venv', '.venv']
+ANCHOR = Path(Path.cwd().anchor)
+
+
+def venv_search(prefix: Path = Path('.')) -> Path:
+    prefix = prefix.absolute()
+
+    found_venvs = []
+
+    for venv in VENV_NAMES:
+        found_venvs.extend(prefix.glob(f"{venv}/bin/python3"))
+
+    if any(found_venvs):
+        #  If there are multiple venvs just return the first one
```

Comment:
Perhaps this ought to be highlighted in the readme

Reply:
Yes good point. In those kind of situations it'd be nice to give some feedback to users, but there's no (as far as I can tell) simple mechanism to raise kernel initialisation warnings that built into Jupyter Notebook/Lab.

Reply:
Added a note about this to the readme